### PR TITLE
[ESKL-59/game-notification] - Game finished notification

### PR DESCRIPTION
**Pull request by nnelas**
**What does this PR do?**

- Adds a new Service that will start whenever an e-Skills game starts.
There are 2 states for this notification: 1) game is playing, so it
shows the players of the room; 2) game is finished, so it shows if the
user has won or lost and how much has won. Nevertheless, there's a
DISMISS button on both states that stops the service and removes the
notification
- Besides that, fixes a typo a string: connection_error_body

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] skills/src/main/java/cm/aptoide/skills/SkillsFragment.kt

**How should this be manually tested?**

  Flow on how to test this or QA Tickets related to this use-case: [ESKL-59](https://aptoide.atlassian.net/browse/ESKL-59)

**What are the relevant tickets?**

  Tickets related to this pull-request: [ESKL-59](https://aptoide.atlassian.net/browse/ESKL-59)

**Questions:**

N/A

**Code Review Checklist**

- [ ] Architecture
- [ ] Documentation on public interfaces
- [ ] Database changed?
- [ ] If yes - Database migration?
- [ ] Remove comments & unused code & forgotten testing Logs
- [ ] Codestyle
- [ ] Functional QA tests pass

### CODE REVIEW

File: skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+        .build()
+  }
+
+  private fun createNotificationDismissIntent(): PendingIntent {
+    val intent = Intent(this, BackgroundGameService::class.java)
+    intent.action = ACTION_DISMISS
+    return PendingIntent.getService(this, 0, intent, 0)
+  }
+
+  override fun onUpdate(roomResponse: RoomResponse) {
```
Comment (Trinkes) [inline skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt:99]: if you only need a string (provided by `getFormattedGameDetails` method) why are you passing the hole network model to view?

File: skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt [hunk 2]
```diff
@@ -0,0 +1,145 @@
...
+      val player: User = roomResponse.users[i]
+      gameDetails.append(player.userName)
+      if (i < roomResponse.users.size - 1) {
+        gameDetails.append(PLAYER_SEPARATOR)
+      }
+    }
+    return gameDetails.toString()
+  }
+
+  override fun onFinishGame(roomResponse: RoomResponse) {
```
Comment (Trinkes) [inline skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt:118]: same as onUpdate

**Review by Trinkes — CHANGES_REQUESTED** [review]

File: skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt [hunk 3]
```diff
@@ -0,0 +1,145 @@
...
+        .setSmallIcon(R.drawable.ic_launcher_foreground)
+        .setDeleteIntent(dismissIntent)
+        .setContentText(text)
+        .build()
+  }
+
+  private fun createNotificationDismissIntent(): PendingIntent {
+    val intent = Intent(this, BackgroundGameService::class.java)
+    intent.action = ACTION_DISMISS
+    return PendingIntent.getService(this, 0, intent, 0)
```
[the remark this answers is not in the document]
Comment (nnelas) [reply skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt:96]: yeap, you're right

File: skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt (hunk 1 above)
Comment (nnelas) [reply skills/src/main/java/cm/aptoide/skills/games/BackgroundGameService.kt]: you're right. changed.

File: skills/src/main/java/cm/aptoide/skills/games/PeriodicGameChecker.kt [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+    private val gameStateListener: GameStateListener) {
+  private val disposables = CompositeDisposable()
+
+  fun start(session: String) {
+    disposables.add(Observable.interval(0, getRoomPeriodSeconds, TimeUnit.SECONDS)
+        .flatMapSingle<Any> {
+          roomRepository.getRoom(session)
+              .observeOn(AndroidSchedulers.mainThread())
+              .doOnSuccess { roomResponse: RoomResponse -> checkGameStatus(roomResponse) }
+              .doOnError(Throwable::printStackTrace)
```
Comment (Trinkes) [inline skills/src/main/java/cm/aptoide/skills/games/PeriodicGameChecker.kt:24]: this will crash if there is a network error

**Review by Trinkes — CHANGES_REQUESTED** [review]

**Pull request merged**